### PR TITLE
CI: fix upload-artifact

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

While inspection it turned out that failing tests were not uploading reports
for example https://github.com/PyTorchLightning/pytorch-lightning/actions/runs/1857677830

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @carmocca @akihironitta @borda

### CODE REVIEW

File: .github/workflows/ci_test-slow.yml [hunk 1]
```diff
@@ -57,15 +57,15 @@ jobs:
...
       env:
         PL_RUN_SLOW_TESTS: 1
 
     - name: Upload pytest test results
       uses: actions/upload-artifact@v2
       with:
-        name: pytest-results-${{ runner.os }}-${{ matrix.python-version }}
-        path: junit/test-results-${{ runner.os }}-${{ matrix.python-version }}.xml
+        name: pytest-results-${{ runner.os }}-py${{ matrix.python-version }}
+        path: junit/test-results-${{ runner.os }}-py${{ matrix.python-version }}.xml
```
Comment (carmocca) [inline .github/workflows/ci_test-slow.yml:68]: does this need?:

```suggestion
        path: junit/test-results-${{ runner.os }}-py${{ matrix.python-version }}.xml
        if-no-files-found: error
```

**Review by carmocca — APPROVED** [review]
Great catch!

**Review by akihironitta — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]

File: .github/workflows/ci_test-slow.yml (hunk 1 above)
Comment (Borda) [reply .github/workflows/ci_test-slow.yml:68]: it shall prevent silent missing reports...

Comment (carmocca) [reply .github/workflows/ci_test-slow.yml:68]: I'm asking if we should add it here too

Comment (Borda) [reply .github/workflows/ci_test-slow.yml:68]: I would say yes

Comment (carmocca) [reply .github/workflows/ci_test-slow.yml:68]: Committing my suggestion then

**Pull request merged**